akka-http-core/src/main/scala/akka/http/Http.scala

```diff
@@ -173,6 +178,208 @@ class HttpExt(config: Config)(implicit system: ActorSystem) extends akka.actor.E
                   settings: ClientConnectionSettings,
                   log: LoggingAdapter = system.log): ClientLayer =
     BidiFlow.wrap(OutgoingConnectionBlueprint(remoteAddress, settings, log))
+
+  /**
+   * Starts a new connection pool to the given host and configuration and returns a [[Flow]] which dispatches
+   * the requests from all its materializations across this pool.
```

Comment:
Yes. Each request creates a new response promise for itself which it `mapAsync`s against. So all responses fulfill the response future of their respective request. The materializations only provide the context for the `mapAsync` stages (and their `parallelism` scope).